src/ParamConverter/DeveloperWithUser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\ParamConverter;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\ParamConverter\ParamConverterInterface;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Resolves "developer_with_user" type parameters in routes.
+ *
+ * Use this parameter converter instead of paramconverter.entity if you would
+ * like to ensure that a developer (by UUID or email address) from Apigee Edge
+ * has a user in Drupal.
+ */
+final class DeveloperWithUser implements ParamConverterInterface {
+
+  /**
+   * The developer entity storage.
+   *
+   * @var \Drupal\apigee_edge\Entity\Storage\DeveloperStorageInterface
+   */
+  private $developerStorage;
+
+  /**
+   * DeveloperWithUser constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->developerStorage = $entity_type_manager->getStorage('developer');
```

Comment:
This causes all of the functional tests in the `apigee_m10n` module to fail with the following error:

```
Drupal\Component\Plugin\Exception\PluginNotFoundException : The "developer" entity type does not exist.

.../core/lib/Drupal/Core/Entity/EntityTypeManager.php:133
.../core/lib/Drupal/Core/Entity/EntityTypeManager.php:234
.../core/lib/Drupal/Core/Entity/EntityTypeManager.php:170
.../modules/contrib/apigee_edge/src/ParamConverter/DeveloperWithUser.php:50
.../core/lib/Drupal/Component/DependencyInjection/Container.php:262
```

Are all of the tests in the `apigee_edge` module still passing with this addition?

Reply:
No, they do not, but Peter has fixed this problem already in his machine and it will be pushed today. I did not expect that you are using a draft branch in M10N module.
Fun fact, the fix is simple. the `$entity_type_manager->getStorage()` can not be used in param converter constructors because entity type manager service collects the storages later than param converters are being built. This may work if I would convert the param converter to a lazy service, not sure.